# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/linux-firmware/used-only.diff
# Copyright (C) 2023 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- linux-firmware-20230625/copy-firmware.sh.vanilla	2023-07-15 18:42:22.097695823 +0200
+++ linux-firmware-20230625/copy-firmware.sh	2023-07-15 18:43:31.595894842 +0200
@@ -70,6 +70,7 @@
 
 # shellcheck disable=SC2162 # file/folder name can include escaped symbols
 grep -E '^(RawFile|File):' WHENCE | sed -E -e 's/^(RawFile|File): */\1 /;s/"//g' | while read k f; do
+    grep -q "^$f$" INSTALLED || continue
     test -f "$f" || continue
     install -d "$destdir/$(dirname "$f")"
     $verbose "copying/compressing file $f$compext"
@@ -71,6 +72,7 @@
 
 # shellcheck disable=SC2162 # file/folder name can include escaped symbols
 grep -E '^Link:' WHENCE | sed -e 's/^Link: *//g;s/-> //g' | while read f d; do
+    grep -q "^$f$" INSTALLED || continue
     if test -L "$f$compext"; then
         test -f "$destdir/$f$compext" && continue
         $verbose "copying link $f$compext"
